### ISSUE TITLE
Can I use `d3-organization-chart` in reactjs?

### ISSUE THREAD

**Issue by hungbang** [bumbeishvili/d3-organization-chart#5]
I am developing the project using Reactjs and Redux. In my platform, we need to use the `d3-organization chart` to show for the company tree to the user. 

Integrating this library to react and the data would be loaded from Backend APIs.

**Comment by bumbeishvili** [comment]
Yes, you will need to create react component and import  TreeChart  into it



```javascript
class OrgChart extends React.Component {
	constructor(props) {
		super(props);
		this.createOrgChartDiagram = this.createOrgChartDiagram.bind(this);
	}

	componentDidMount() {
		this.createOrgChartDiagram();
	}

	componentDidUpdate(prevProps, prevState) {
		this.createOrgChartDiagram();
	}

	render() {
		return (
			<div>
				<div ref={(node) => (this.node = node)} />
			</div>
		);
	}

	createOrgChartDiagram() {
		const node = this.node;

		if (!this.chart) {
			this.chart =new TreeChart()
				.container(node)
				.data(this.props.data)
				.svgHeight(window.innerHeight - 30)
				.svgWidth(window.innerWidth - 30)
                                 .initialZoom(0.4)
                                 .onNodeClick(d=> console.log(d+' node clicked'))
                                 .render()
		} else {
			this.chart.data(this.props.data)
			      .render();
		}
	}
}

export default OrgChart;
```

Then you can use it as
```react
	<div>
		<OrgChart data={backendOrgChartData} />
	</div>
```

**Comment by hungbang** [comment]
@bumbeishvili Thank you so much. I am very appreciate

**Comment by hungbang** [comment]
![image](https://user-images.githubusercontent.com/10163092/68821653-7d5e2c80-06c1-11ea-8681-0899b0727f82.png)
I got error : 
`ReferenceError: d3 is not defined`

**Comment by bumbeishvili** [comment]
I am not sure what's problem, I havent used treeChart in react by myself. 

Once I will have time, I will provide demo react , angular and vue projects using this orgChart, though currently I don't know how can I help you, since I have no availability to work on this

**Comment by hungbang** [comment]
@bumbeishvili I see, I guess we need to help from the community for this. Anyway, thank you so much for your time.

**Comment by hungbang** [comment]
I found the similar from here: https://react-org-chart.now.sh/examples/simple , please take a look, I hope It is able to help. @bumbeishvili

**Comment by bumbeishvili** [comment]
Thanks for the understanding , will take a look , thank you for the link

**Comment by bumbeishvili** [comment]
@hungbang  I think I fixed that particular issue

I was trying to integrate visualization within vue and got similar error.  

I believe code will stay same  as wrote above in response of your issue.  Just make  sure, that you import org chart as follows.

`import * as TreeChart from "d3-org-chart";`

I don't know if you are familiar with vue, but I will link relevant file anyway, it may help you.

https://github.com/bumbeishvili/d3-organization-chart-integrations/blob/master/vue/org-chart-demo/src/components/Chart.vue#L8

**Comment by hungbang** [comment]
Thank you so much for your sample. I've very appreciated 🥇 
Let me try again with this. If it is able to work properly, I will give you the sample `d3-organization-chart-integrations`. Maybe it might help the community

**Comment by bumbeishvili** [comment]
Just a note - I created vue, react and angular integration demos

[Integrations](https://github.com/bumbeishvili/d3-organization-chart/blob/master/README.md#integrations)
